File: TOMIMSV4/Entity/Transacciones/Despacho/clsBeTrans_despacho_enc.vb
Public Class clsBeTrans_despacho_enc
    Implements ICloneable
    Implements IDisposable

    Public Property IdDespachoEnc() As Integer = 0
    Public Property IdBodega() As Integer = 0
    Public Property IdPropietarioBodega() As Integer = 0
    Public Property IdVehiculo() As Integer = 0
    Public Property IdPiloto() As Integer = 0
    Public Property IdRuta() As Integer = 0
    Public Property Fecha() As DateTime = Date.Now
    Public Property No_pase() As Integer = 0
    Public Property Observacion() As String = ""
    Public Property Hora_ini() As DateTime
    Public Property Hora_fin() As Date = Date.Now
    Public Property Estado() As String = ""
    Public Property Numero() As Double = 0
    Public Property Marchamo() As String = ""
    Public Property Cant_bultos() As Double = 0.0
    Public Property User_agr() As String = ""
    Public Property Fec_agr() As Date = Date.Now
    Public Property User_mod() As String = ""
    Public Property Fec_mod() As Date = Date.Now
    Public Property Activo() As Boolean = False
    Public Property No_Documento_Externo As String = ""

    Sub New()
    End Sub

    Sub New(ByRef IdDespachoEnc As Integer, ByVal IdBodega As Integer, ByVal IdPropietarioBodega As Integer, ByVal IdVehiculo As Integer, ByVal IdPiloto As Integer, ByVal IdRuta As Integer, ByVal fecha As Date, ByVal no_pase As Integer, ByVal observacion As String, ByVal hora_ini As Date, ByVal hora_fin As Date, ByVal estado As String, ByVal numero As Integer, ByVal marchamo As String, ByVal cant_bultos As Double, ByVal user_agr As String, ByVal fec_agr As Date, ByVal user_mod As String, ByVal fec_mod As Date, ByVal activo As Boolean)
        Me.IdDespachoEnc = IdDespachoEnc
        Me.IdBodega = IdBodega
        Me.IdPropietarioBodega = IdPropietarioBodega
        Me.IdVehiculo = IdVehiculo
        Me.IdPiloto = IdPiloto
        Me.IdRuta = IdRuta
        Me.Fecha = Fecha
        Me.No_pase = No_pase
        Me.Observacion = Observacion
        Me.Hora_ini = Hora_ini
        Me.Hora_fin = Hora_fin
        Me.Estado = Estado
        Me.Numero = Numero
        Me.Marchamo = Marchamo
        Me.Cant_bultos = Cant_bultos
        Me.User_agr = User_agr
        Me.Fec_agr = Fec_agr
        Me.User_mod = User_mod
        Me.Fec_mod = Fec_mod
        Me.Activo = Activo
    End Sub

    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

End Class

File: WMSWebAPI/Entity/Despacho/clsBeTrans_despacho_enc.cs
using WMSWebAPI.Be;

namespace WMSWebAPI.Entity.Despacho
{
    public class clsBeTrans_despacho_enc : ICloneable, IDisposable
    {
        public int IdDespachoEnc { get; set; } = 0;
        public int IdBodega { get; set; } = 0;
        public int IdPropietarioBodega { get; set; } = 0;
        public int IdVehiculo { get; set; } = 0;
        public int IdPiloto { get; set; } = 0;
        public int IdRuta { get; set; } = 0;
        public DateTime Fecha { get; set; } = DateTime.Now;
        public int No_pase { get; set; } = 0;
        public string Observacion { get; set; } = string.Empty;
        public DateTime Hora_ini { get; set; } = DateTime.MinValue;
        public DateTime Hora_fin { get; set; } = DateTime.Now;
        public string Estado { get; set; } = string.Empty;
        public double Numero { get; set; } = 0;
        public string Marchamo { get; set; } = string.Empty;
        public double Cant_bultos { get; set; } = 0.0;
        public string User_agr { get; set; } = string.Empty;
        public DateTime Fec_agr { get; set; } = DateTime.Now;
        public string User_mod { get; set; } = string.Empty;
        public DateTime Fec_mod { get; set; } = DateTime.Now;
        public bool Activo { get; set; } = false;
        public string No_Documento_Externo { get; set; } = string.Empty;
        public List<clsBeTrans_despacho_det> Detalle { get; set; } = new List<clsBeTrans_despacho_det>();
        public clsBeTrans_despacho_enc() { }

        public clsBeTrans_despacho_enc(
            int idDespachoEnc, int idBodega, int idPropietarioBodega, int idVehiculo, int idPiloto, int idRuta,
            DateTime fecha, int no_pase, string observacion, DateTime hora_ini, DateTime hora_fin,
            string estado, double numero, string marchamo, double cant_bultos,
            string user_agr, DateTime fec_agr, string user_mod, DateTime fec_mod, bool activo)
        {
            IdDespachoEnc = idDespachoEnc;
            IdBodega = idBodega;
            IdPropietarioBodega = idPropietarioBodega;
            IdVehiculo = idVehiculo;
            IdPiloto = idPiloto;
            IdRuta = idRuta;
            Fecha = fecha;
            No_pase = no_pase;
            Observacion = observacion;
            Hora_ini = hora_ini;
            Hora_fin = hora_fin;
            Estado = estado;
            Numero = numero;
            Marchamo = marchamo;
            Cant_bultos = cant_bultos;
            User_agr = user_agr;
            Fec_agr = fec_agr;
            User_mod = user_mod;
            Fec_mod = fec_mod;
            Activo = activo;
        }

        public object Clone()
        {
            return MemberwiseClone();
        }

        public void Dispose()
        {
            // Recurso administrado liberado manualmente (si aplica en futuro)
            GC.SuppressFinalize(this);
        }
    }

}
